deepdrivemd/models/keras_cvae_stream/train.py

```diff
@@ -0,0 +1,168 @@
+    """
+
+    # Wait for enough bpfiles
+    while True:
+        bpfiles = glob.glob(str(cfg.agg_dir / "*/*/agg.bp"))
+        if len(bpfiles) == cfg.num_agg:
+            break
+        print(f"Waiting for {cfg.num_agg} agg.bp files")
+        time.sleep(cfg.timeout1)
+
+    print(f"bpfiles = {bpfiles}")
+
+    # Wait for enough time steps in each bp file
+    while True:
+        enough = True
+        for bp in bpfiles:
+            com = f"bpls {bp}"
+            a = subprocess.getstatusoutput(com)
+            if a[0] != 0:
+                enough = False
+                print(f"Waiting, a = {a}, {bp}")
+                break
+            try:
+                steps = int(a[1].split("\n")[0].split("*")[0].split(" ")[-1])
+            except Exception as e:
+                print("Exception ", e)
+                steps = 0
+                enough = False
+            if steps < cfg.min_step_increment:
+                enough = False
+                print(f"Waiting, steps = {steps}, {bp}")
+                break
+        if enough:
+            break
+        else:
+            time.sleep(cfg.timeout2)
+
+    return bpfiles
+
+
+def next_input(
+    cfg: KerasCVAEModelConfig, streams: Streams
+) -> Tuple[np.ndarray, np.ndarray]:
+    """Read the next batch of contact maps from aggregated files.
+
+    Returns
+    -------
+    Tuple[np.ndarray, np.ndarray]
+          Training and validation sets.
+    """
+
+    with Timer("ml_read"):
+        cm_data_input = streams.next()[0]
+    cm_data_input = np.expand_dims(cm_data_input, axis=-1)
+    print(
+        f"in next_input: cm_data_input.shape = {cm_data_input.shape}"
+    )  # (2000, 28, 28, 1)
+    np.random.shuffle(cm_data_input)
+    train_val_split = int(0.8 * len(cm_data_input))
+    print(f"train_val_split = {train_val_split}")
+    sys.stdout.flush()
+    return cm_data_input[:train_val_split], cm_data_input[train_val_split:]
+
+
+def build_model(cfg: KerasCVAEModelConfig):
+    with Timer("ml_conv_variational_autoencoder"):
+        cvae = CVAE(
+            image_size=cfg.final_shape,
+            channels=cfg.final_shape[-1],
+            conv_layers=cfg.conv_layers,
+            feature_maps=cfg.conv_filters,
+            filter_shapes=cfg.conv_filter_shapes,
+            strides=cfg.conv_strides,
+            dense_layers=cfg.dense_layers,
+            dense_neurons=cfg.dense_neurons,
+            dense_dropouts=cfg.dense_dropouts,
+            latent_dim=cfg.latent_dim,
+        )
+    cvae.model.summary()
+    return cvae
+
+
+def main(cfg: KerasCVAEModelConfig):
+    print(subprocess.getstatusoutput("hostname")[1])
+    sys.stdout.flush()
+
+    cfg.checkpoint_dir = cfg.output_path / "checkpoints"
+    cfg.checkpoint_dir.mkdir(exist_ok=True)
+
+    cfg.published_model_dir = cfg.output_path / "published_model"
+    cfg.published_model_dir.mkdir(exist_ok=True)
+
+    with Timer("ml_wait_for_input"):
+        cfg.bpfiles = wait_for_input(cfg)
```

Comment:
Can bpfiles simply be dropped into a local variable? It doesn't seem like it needs to be part of the config. Also, it is currently missing from the config and would have to be added -- which doesn't seem like the right solution since it is only passed to Streams.

Proposed solution:
```
with Timer("ml_wait_for_input"):
        bpfiles = wait_for_input(cfg)
        
 streams = Streams(
        bpfiles,
        [StreamContactMapVariable("contact_map", np.uint8, 1)],
        lastN=cfg.max_steps,
        config=cfg.adios_xml_agg,
        batch=cfg.read_batch,
    )
```

Reply:
Done